pkg/agent/monitortool/monitor.go

```diff
@@ -0,0 +1,493 @@
+	isIPv4Enabled bool
+	// isIPv6Enabled is the flag to indicate if the IPv6 is enabled.
+	isIPv6Enabled bool
+
+	// The map of node name to node info, it will changed by node watcher
+	nodeInformer coreinformers.NodeInformer
+	// nodeLatencyMonitorInformer is the informer for the NodeLatencyMonitor CRD.
+	nodeLatencyMonitorInformer crdinformers.NodeLatencyMonitorInformer
+}
+
+// LatencyConfig is the config for the latency monitor.
+type LatencyConfig struct {
+	// Enable is the flag to enable the latency monitor.
+	Enable bool
+	// Interval is the interval time to ping all nodes.
+	Interval time.Duration
+}
+
+func NewNodeLatencyMonitor(nodeInformer coreinformers.NodeInformer,
+	nlmInformer crdinformers.NodeLatencyMonitorInformer,
+	nodeConfig *config.NodeConfig,
+	trafficEncapMode config.TrafficEncapModeType) *NodeLatencyMonitor {
+	m := &NodeLatencyMonitor{
+		nodeConfig:                 nodeConfig,
+		trafficEncapMode:           trafficEncapMode,
+		latencyStore:               NewLatencyStore(trafficEncapMode.IsNetworkPolicyOnly()),
+		latencyConfig:              &LatencyConfig{Enable: false},
+		latencyConfigChanged:       make(chan struct{}, 1),
+		nodeInformer:               nodeInformer,
+		nodeLatencyMonitorInformer: nlmInformer,
+	}
+
+	// Get the IPv4/IPv6 enabled status
+	isIPv4Enabled, err := config.IsIPv4Enabled(m.nodeConfig, m.trafficEncapMode)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get IPv4 enabled status")
+	}
+	isIPv6Enabled, err := config.IsIPv6Enabled(m.nodeConfig, m.trafficEncapMode)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get IPv6 enabled status")
+	}
+	m.isIPv4Enabled = isIPv4Enabled
+	m.isIPv6Enabled = isIPv6Enabled
+
+	// Add node informer event handler for Node
+	nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    m.onNodeAdd,
+		UpdateFunc: m.onNodeUpdate,
+		DeleteFunc: m.onNodeDelete,
+	})
+
+	// Add crd informer event handler for NodeLatencyMonitor
+	nlmInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    m.onNodeLatencyMonitorAdd,
+		UpdateFunc: m.onNodeLatencyMonitorUpdate,
+		DeleteFunc: m.onNodeLatencyMonitorDelete,
+	})
+
+	return m
+}
+
+// onNodeAdd is the event handler for adding Node.
+func (m *NodeLatencyMonitor) onNodeAdd(obj interface{}) {
+	node := obj.(*corev1.Node)
+	m.latencyStore.addNode(node)
+
+	klog.InfoS("Node added", "Node", klog.KObj(node))
+}
+
+// onNodeUpdate is the event handler for updating Node.
+func (m *NodeLatencyMonitor) onNodeUpdate(oldObj, newObj interface{}) {
+	oldNode := oldObj.(*corev1.Node)
+	node := newObj.(*corev1.Node)
+	m.latencyStore.updateNode(oldNode, node)
+
+	klog.InfoS("Node updated", "Node", klog.KObj(node))
+}
+
+// onNodeDelete is the event handler for deleting Node.
+func (m *NodeLatencyMonitor) onNodeDelete(obj interface{}) {
+	// Check if the object is a not a node
+	node, ok := obj.(*corev1.Node)
+	if !ok {
+		return
+	}
```

Comment:
delete event handlers usually need to be a bit more complex. Please refer to: https://github.com/antrea-io/antrea/blob/572790d207ef8da205efb12b2c497809fdf96a2a/pkg/agent/controller/egress/ip_scheduler.go#L154-L170

Reply:
in your case however, you should use `klog.ErrorS` and not `klog.Errorf`

Reply:
Ok, I skip the `DeletedFinalStateUnknown` type before, I will update it.